solver/src/analytics.rs

```diff
@@ -0,0 +1,126 @@
+use crate::{metrics::SolverMetrics, settlement::Settlement};
+use bigdecimal::{ToPrimitive, Zero};
+use ethcontract::H160;
+use model::order::OrderUid;
+use num::BigRational;
+use shared::conversions::U256Ext;
+use std::{
+    collections::{hash_map::Entry, HashMap},
+    fmt::{Display, Formatter},
+    sync::Arc,
+};
+
+#[derive(Clone)]
+struct SurplusInfo {
+    solver_name: &'static str,
+    absolute: BigRational,
```

Comment:
I think we should get rid of this property for now since we shouldn't use it (since it isn't normalized at all).